pdoc/docstrings.py

```diff
@@ -366,11 +366,48 @@ def _rst_admonitions(contents: str, source_file: Path | None) -> str:
     <https://www.sphinx-doc.org/en/master/usage/restructuredtext/directives.html>
     """
 
+    def _rst_extract_options(contents: str) -> tuple[str, dict[str, str]]:
+        """
+        Extract options from the beginning of reStructuredText directives.
+        """
+        matches = re.finditer(r"\s*:(.+?):\s*?(.*?)$", contents, re.MULTILINE)
```

Comment:
Two suggestions here:

1. Options should only appear at the start, not in the middle of the docstring. So this should probable be a `while m := re.match(...):` loop that checks if there currently is an option at the start, processes it, and chops it off.
2. The regex currently isn't great with multiple non-greedy groups next to each other. For example, the second `\s*?` will never match anything. Proposal for an alternative that works with 1):
   ```regex
    ^\s*:(.+?):[ \t]*(.*)([\s\S]*)
   ```
   The last capture group is the remainder of `contents`.

Reply:
Thanks for the tips! That made the function much neater